libfuse/mounter.go

```diff
@@ -87,17 +87,27 @@ func (m ForceMounter) Unmount() (err error) {
 
 func doUnmount(dir string, force bool) (err error) {
 	switch runtime.GOOS {
-	case "darwin":
-		if force {
-			_, err = exec.Command("/usr/sbin/diskutil", "unmountDisk", "force", dir).Output()
-		} else {
-			_, err = exec.Command("/sbin/umount", dir).Output()
+	case "darwin", "linux":
+		switch runtime.GOOS {
+		case "darwin":
+			if force {
+				_, err = exec.Command(
+					"/usr/sbin/diskutil", "unmountDisk", "force", dir).Output()
+			} else {
+				_, err = exec.Command("/sbin/umount", dir).Output()
+			}
+		case "linux":
+			if force {
+				_, err = exec.Command("fusermount", "-ul", dir).Output()
+			} else {
+				_, err = exec.Command("fusermount", "-u", dir).Output()
+			}
 		}
-	case "linux":
-		if force {
-			_, err = exec.Command("fusermount", "-ul", dir).Output()
-		} else {
-			_, err = exec.Command("fusermount", "-u", dir).Output()
+		if err != nil {
```

Comment:
Do you really need a double switch here?  Why not just move this `if` block down below the `switch`?  Seems like it would work just fine in the default case.